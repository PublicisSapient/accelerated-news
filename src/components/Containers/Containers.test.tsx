import React from 'react';
import { render } from '../../test/test-utils';
import {
  CenteredContainer,
  HorizontalContainer,
  ScrollingContainer,
  VerticalContainer,
  ViewCenteredContainer,
  ViewHorizontalContainer,
  ViewVerticalContainer,
} from './Containers';

describe('Containers', () => {
  test('HorizontalContainer renders correctly', () => {
    const { asFragment } = render(
      <HorizontalContainer>Hello World!</HorizontalContainer>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('ViewHorizontalContainer renders correctly', () => {
    const { asFragment } = render(
      <ViewHorizontalContainer>Hello World!</ViewHorizontalContainer>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('VerticalContainer renders correctly', () => {
    const { asFragment } = render(
      <VerticalContainer>Hello World!</VerticalContainer>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('ViewVerticalContainer renders correctly', () => {
    const { asFragment } = render(
      <ViewVerticalContainer>Hello World!</ViewVerticalContainer>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('CenteredContainer renders correctly', () => {
    const { asFragment } = render(
      <CenteredContainer>Hello World!</CenteredContainer>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('ViewCenteredContainer renders correctly', () => {
    const { asFragment } = render(
      <ViewCenteredContainer>Hello World!</ViewCenteredContainer>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('ScrollingContainer renders correctly', () => {
    const { asFragment } = render(
      <ScrollingContainer>Hello World!</ScrollingContainer>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
