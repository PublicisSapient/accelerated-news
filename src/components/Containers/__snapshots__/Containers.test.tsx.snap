// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers CenteredContainer renders correctly 1`] = `
<DocumentFragment>
  <div
    class="flex-1 flex justify-center items-center text-center "
    data-testid="centered-container"
  >
    Hello World!
  </div>
</DocumentFragment>
`;

exports[`Containers HorizontalContainer renders correctly 1`] = `
<DocumentFragment>
  <div
    class="flex-1 flex flex-row "
    data-testid="horizontal-container"
  >
    Hello World!
  </div>
</DocumentFragment>
`;

exports[`Containers ScrollingContainer renders correctly 1`] = `
<DocumentFragment>
  <div
    class="overflow-auto "
    data-testid="scrolling-container"
  >
    Hello World!
  </div>
</DocumentFragment>
`;

exports[`Containers VerticalContainer renders correctly 1`] = `
<DocumentFragment>
  <div
    class="flex-1 flex flex-col "
    data-testid="vertical-container"
  >
    Hello World!
  </div>
</DocumentFragment>
`;

exports[`Containers ViewCenteredContainer renders correctly 1`] = `
<DocumentFragment>
  <div
    class="h-screen flex justify-center items-center text-center "
    data-testid="centered-container"
  >
    Hello World!
  </div>
</DocumentFragment>
`;

exports[`Containers ViewHorizontalContainer renders correctly 1`] = `
<DocumentFragment>
  <div
    class="h-screen flex flex-row "
    data-testid="view-horizontal-container"
  >
    Hello World!
  </div>
</DocumentFragment>
`;

exports[`Containers ViewVerticalContainer renders correctly 1`] = `
<DocumentFragment>
  <div
    class="h-screen flex flex-col "
    data-testid="view-vertical-container"
  >
    Hello World!
  </div>
</DocumentFragment>
`;
